arangod/Cluster/ClusterMethods.cpp

```diff
@@ -1083,47 +1083,6 @@ futures::Future<Result> warmupOnCoordinator(ClusterFeature& feature,
   if (collinfo == nullptr) {
     return futures::makeFuture(Result(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND));
   }
-  std::vector<std::string> shards;
-
-  auto addShards = [](std::vector<std::string>& result,
-                      std::shared_ptr<LogicalCollection> const& collection) {
-    if (collection == nullptr) {
-      return;
-    }
-    auto shardIds = collection->shardIds();
-    for (auto const& it : *shardIds) {
-      result.emplace_back(it.first);
-    }
-  };
-
-  addShards(shards, collinfo);
-
-#ifdef USE_ENTERPRISE
-  if (collinfo->isSmart() && collinfo->type() == TRI_COL_TYPE_EDGE) {
-    // SmartGraph galore
-    auto theEdge = dynamic_cast<arangodb::VirtualClusterSmartEdgeCollection*>(
-        collinfo.get());
-    if (theEdge != nullptr) {
-      CollectionNameResolver resolver(collinfo->vocbase());
-
-      std::string name =
-          resolver.getCollectionNameCluster(theEdge->getLocalCid());
-      auto c = ci.getCollectionNT(dbname, name);
-      addShards(shards, c);
-
-      name = resolver.getCollectionNameCluster(theEdge->getFromCid());
-      c = ci.getCollectionNT(dbname, name);
-      addShards(shards, c);
-
-      name = resolver.getCollectionNameCluster(theEdge->getToCid());
-      c = ci.getCollectionNT(dbname, name);
-      addShards(shards, c);
-    }
-  }
-#endif
-  // now make shards in vector unique
-  std::sort(shards.begin(), shards.end());
-  shards.erase(std::unique(shards.begin(), shards.end()), shards.end());
```

Comment:
Can you clarify why this is no longer necessary?

Reply:
I added this in 3.9, because in 3.9 it was necessary.
It turned out that in devel someone already made a change which makes the shards of a SmartGraph edge collection available via the normal APIs, i.e. no further workaround is necessary here.
However, breaking news as of this noon: that change I just referred to just got reverted in devel and 3.10.
So I have to check if I need to add this back or not.